server/sonar-server/src/main/java/org/sonar/server/health/NodeHealthProviderImpl.java

```diff
@@ -0,0 +1,99 @@
+ * SonarQube
+ * Copyright (C) 2009-2017 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.server.health;
+
+import java.util.Random;
+import java.util.function.Supplier;
+import org.sonar.api.config.Configuration;
+import org.sonar.api.platform.Server;
+import org.sonar.api.utils.System2;
+import org.sonar.cluster.health.NodeDetails;
+import org.sonar.cluster.health.NodeHealth;
+import org.sonar.cluster.health.NodeHealthProvider;
+
+import static java.lang.String.format;
+import static org.sonar.cluster.ClusterProperties.CLUSTER_NAME;
+import static org.sonar.cluster.ClusterProperties.CLUSTER_PORT;
+import static org.sonar.cluster.ClusterProperties.CLUSTER_SEARCH_DISABLED;
+import static org.sonar.cluster.health.NodeDetails.newNodeDetailsBuilder;
+import static org.sonar.cluster.health.NodeHealth.newNodeHealthBuilder;
+
+public class NodeHealthProviderImpl implements NodeHealthProvider {
+  private final HealthChecker healthChecker;
+  private final NodeHealth.Builder nodeHealthBuilder;
+  private final NodeDetails nodeDetails;
+  private final System2 system2;
+
+  public NodeHealthProviderImpl(Configuration configuration, HealthChecker healthChecker, Server server, System2 system2) {
+    this.system2 = system2;
+    this.healthChecker = healthChecker;
+    this.nodeHealthBuilder = newNodeHealthBuilder();
+    this.nodeDetails = newNodeDetailsBuilder()
+      .setName(computeName(configuration))
+      .setType(computeType(configuration))
+      .setHost(computeIp(configuration))
+      .setPort(computePort(configuration))
+      .setStarted(server.getStartedAt().getTime())
+      .build();
+  }
+
+  private static String computeName(Configuration configuration) {
+    String clusterName = configuration.get(CLUSTER_NAME)
+      .orElseThrow(missingPropertyISE(CLUSTER_NAME));
+    return clusterName + new Random().nextInt(999);
+  }
+
+  private static NodeDetails.Type computeType(Configuration configuration) {
+    if (configuration.getBoolean(CLUSTER_SEARCH_DISABLED).orElse(false)) {
+      return NodeDetails.Type.APPLICATION;
+    }
+    return NodeDetails.Type.SEARCH;
+  }
+
+  private static String computeIp(Configuration configuration) {
+    return "hardcoded IP";
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this method and declare a constant for this value. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS3400)